### DISCUSSION TITLE
add fetchPolicy option to useQuery hook

### DISCUSSION THREAD

**Discussion by bebjakub** [gqty-dev/gqty#913] [Ideas]
Hi :)
During the implementation of [On-demand Incremental Static Regeneration](https://github.com/wpengine/faustjs/discussions/842), I experienced issues with duplicates of list items after calling unstable_revalidate.

After getting list items simply via fetch I'm not experiencing the issues anymore.

Is there a possibility to add fetchPolicy option with 'no-cache' acceptable to useQuery hook?

On the Discord I got suggestions to fetch with useLazyQuery, but there are many scenarios when async calls are not acceptable https://github.com/wpengine/faustjs/blob/canary/examples/next/getting-started/src/pages/posts/index.tsx#L18 or it'll be just easier to define it in the useQuery hook like looking for other unnecessary workarounds.

Thank you!

**Comment by vicary** [comment]
I am picking up old conversations here.

Fetch policy in `useQuery()` is within my roadmap, stay tuned.

Meanwhile I'd like to understand more about your use case, because `no-cache` means a fetch, how does it work if async calls are not acceptable?

**Comment by bebjakub** [comment]
Hi @vicary it's not an issue anymore, faust.js had a huge refactor.

**Reply by vicary** [reply]
That's good to know!
